### PR TITLE
Remove namespace for CustomResourceDefinition in CreateAndWaitForCRD

### LINKED ISSUE #546: Remove namespace for CustomResourceDefinition in CreateAndWaitForCRD
CustomResourceDefinition is not namespace'd object and namespace is not needed during Get and Create in tests

### PR DESCRIPTION
**Pull request by dmvolod**
Fixes #546

### CODE REVIEW

File: test/e2e/k8s/kubernetes.go [hunk 1]
```diff
@@ -204,19 +204,18 @@ func (k TestKubernetes) GracefulRestartInfinispan(infinispan *ispnv1.Infinispan,
...
 
 // CreateAndWaitForCRD creates a Custom Resource Definition, waiting it to become ready.
-func (k TestKubernetes) CreateAndWaitForCRD(crd *apiextv1beta1.CustomResourceDefinition, namespace string) {
-	fmt.Printf("Create CRD %s\n", crd.ObjectMeta.Name)
+func (k TestKubernetes) CreateAndWaitForCRD(crd *apiextv1beta1.CustomResourceDefinition) {
+	fmt.Printf("Create CRD %s\n", crd.Name)
 
-	ns := types.NamespacedName{Name: crd.ObjectMeta.Name, Namespace: namespace}
-	err := k.Kubernetes.Client.Get(context.TODO(), ns, crd)
+	err := k.Kubernetes.Client.Get(context.TODO(), types.NamespacedName{Name: crd.Name}, crd)
```
Comment (ryanemerson) [inline test/e2e/k8s/kubernetes.go:210]: AFAICT `crd.Name` hasn't been set when `CreateAndWaitForCRD` is called with either the cache or infinispan CRD

Also what happens if a CRD is installed in two different namespaces with the same name?

Comment (dmvolod) [reply test/e2e/k8s/kubernetes.go:210]: But this PR is just a syntax fix, this working fine without changes as client controller ignores Namespace definition.

Comment (ryanemerson) [reply test/e2e/k8s/kubernetes.go:210]: Yes you're right. I'm still getting upto speed with the operator, so thanks for the explanation :+1:

Comment (dmvolod) [reply test/e2e/k8s/kubernetes.go:210]: Your first operator code is excellent, it's much more about Kubernetes. If you have a time, a good point to involve at the kubernetes sigs infra projects like controller-runtime, kubebuilder as well as Operator SDK.

**Comment by ryanemerson** [comment]
Thanks @dmvolod

**Pull request merged**